airflow/providers/common/sql/hooks/sql.py

```diff
@@ -264,6 +266,7 @@ def run(
                 results = []
                 for sql_statement in sql:
                     self._run_command(cur, sql_statement, parameters)
+                    self._update_query_ids(cur)
```

Comment:
i think that we need to address this issue, where the queries aren't actually running and therefore will not be killed, before adding this interface.